src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -768,15 +797,15 @@ fn trait_impls_for<'a>(
 /// Check for resolve collisions between a trait and its derive.
 ///
 /// These are common and we should just resolve to the trait in that case.
-fn is_derive_trait_collision<T>(ns: &PerNS<Result<(Res, T), ResolutionFailure<'_>>>) -> bool {
-    matches!(
-        *ns,
-        PerNS {
-            type_ns: Ok((Res::Def(DefKind::Trait, _), _)),
-            macro_ns: Ok((Res::Def(DefKind::Macro(MacroKind::Derive), _), _)),
-            ..
-        }
-    )
+fn is_derive_trait_collision<T>(ns: &PerNS<Result<Vec<(Res, T)>, ResolutionFailure<'_>>>) -> bool {
+    if let (&Ok(ref type_ns), &Ok(ref macro_ns)) = (&ns.type_ns, &ns.macro_ns) {
+        type_ns.iter().any(|(res, _)| matches!(res, Res::Def(DefKind::Trait, _)))
```

Comment:
I think it's not an issue since we only want to check if there is a derive macro and a trait. In one case, we run this check if we have two elements which then allow to only return the types, and in the other case, we remove the macros from the candidates being returned.

Reply:
It seems the function wasn't needed so I removed it completely.

Reply:
No, it was needed:

```
error: `PartialEq` is both a trait and a derive macro
  --> library/core/src/macros/mod.rs:65:71
   |
65 | /// Asserts that two expressions are not equal to each other (using [`PartialEq`]).
   |                                                                       ^^^^^^^^^ ambiguous link
   |
help: to link to the trait, prefix with `trait@`
   |
65 | /// Asserts that two expressions are not equal to each other (using [`trait@PartialEq`]).
   |                                                                       ++++++
help: to link to the derive macro, prefix with `derive@`
   |
65 | /// Asserts that two expressions are not equal to each other (using [`derive@PartialEq`]).
   |                                                                       +++++++
```

I'll add a regression test.